src/action/macos/create_fstab_entry.rs

```diff
@@ -236,76 +148,67 @@ impl Action for CreateFstabEntry {
     #[tracing::instrument(level = "debug", skip_all)]
     async fn revert(&mut self) -> Result<(), ActionError> {
         let fstab_path = Path::new(FSTAB_PATH);
-
-        if let Some(diskutil_info) = get_disk_info_for_label(&self.apfs_volume_label)
+        let mut fstab = OpenOptions::new()
+            .create(false)
+            .write(true)
+            .read(true)
+            .open(&fstab_path)
             .await
-            .map_err(Self::error)?
-        {
-            let fstab_entry = fstab_lines(&diskutil_info.volume_uuid, &self.apfs_volume_label);
+            .map_err(|e| Self::error(ActionErrorKind::Open(fstab_path.to_owned(), e)))?;
 
-            let mut file = OpenOptions::new()
-                .create(false)
-                .write(true)
-                .read(true)
-                .open(&fstab_path)
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Open(fstab_path.to_owned(), e)))?;
-
-            let mut file_contents = String::default();
-            file.read_to_string(&mut file_contents)
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Read(fstab_path.to_owned(), e)))?;
+        let mut fstab_buf = String::new();
+        fstab
+            .read_to_string(&mut fstab_buf)
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Read(fstab_path.to_owned(), e)))?;
 
-            if let Some(start) = file_contents.rfind(fstab_entry.as_str()) {
-                let end = start + fstab_entry.len();
-                file_contents.replace_range(start..end, "")
-            }
+        let updated_buf = fstab_buf
+            .lines()
+            .filter_map(|line| {
+                // Delete nix-installer entries with a "prelude" comment
+                if line.starts_with("# nix-installer created volume labelled") {
+                    None
+                } else {
+                    Some(line)
+                }
+            })
+            .filter_map(|line| {
+                if line.split(&[' ', '\t']).nth(2) == Some("/nix") {
```

Comment:
I think this has always been wrong, which may have caused our issue..... :|

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0a52701d536dacd5ae262681fd602647

This is looking for the "type" (i.e. apfs, ext4, ....) to match `/nix` lol

Reply:
Fixed!